### PR TITLE
Pass versions to Dockerfile from Makefile

### PR DESCRIPTION
**Pull request by redhug1**
### What

Improve build process to only have versions in Makefile and pass then into Dockerfile during build

### How to review

Inspect, check with doing:
make build

(delete the local image afterwards with: docker rmi <image id>
 and, then run: docker system prune)

### Who can review

Anyone

### CODE REVIEW

File: python-minimal/Makefile [hunk 1]
```diff
@@ -2,13 +2,13 @@ SHELL := bash
...
-PYTHON_VERSION = 3.12
-ALPINE_VERSION = 3.20.3
+PYTHON_VERSION=3.12
+ALPINE_VERSION=3.20.3
 IMAGE_ALIAS = onsdigital/dp-concourse-tools-python-minimal:${ALPINE_VERSION}-python-${PYTHON_VERSION}
 
 .PHONY: build
 build:
-	docker build -t ${IMAGE_ALIAS} -f Dockerfile.alpine-python-minimal .
+	docker build -t ${IMAGE_ALIAS} --build-arg PYTHON_VERSION=${PYTHON_VERSION} --build-arg ALPINE_VERSION="${ALPINE_VERSION}" -f Dockerfile.alpine-python-minimal .
```
Comment (gedge) [inline python-minimal/Makefile:11]:
```suggestion
	docker build -t ${IMAGE_ALIAS} --build-arg PYTHON_VERSION=${PYTHON_VERSION} --build-arg ALPINE_VERSION="${ALPINE_VERSION}" .
```

all the other directories use the default name for the `Dockerfile` - why not use that name in this directory? would make the above command shorter

so, to accompany my suggestion:

`git mv Dockerfile.alpine-python-minimal Dockerfile`

File: ubuntu-20/Makefile [hunk 2]
```diff
@@ -2,12 +2,13 @@ SHELL := bash
...
-UBUNTU_VERSION = ubuntu20.4-focal-20241011
-IMAGE_NAME = ${ECR_AWS_ACCOUNT_URL}/onsdigital/dp-concourse-tools-ubuntu-20:${UBUNTU_VERSION}
+UBUNTU_YEAR_NAME = ubuntu20.4
+UBUNTU_VERSION = focal-20241011
+IMAGE_NAME = ${ECR_AWS_ACCOUNT_URL}/onsdigital/dp-concourse-tools-ubuntu-20:${UBUNTU_YEAR_NAME}-${UBUNTU_VERSION}
 
 .PHONY: build
 build:
-	docker build -t ${IMAGE_NAME} -f Dockerfile .
+	docker build -t ${IMAGE_NAME} --build-arg UBUNTU_VERSION=${UBUNTU_VERSION} .
```
Comment (gedge) [inline ubuntu-20/Makefile:11]:
```suggestion
	docker build -t ${IMAGE_ALIAS} --build-arg UBUNTU_VERSION=${UBUNTU_VERSION} .
```

File: ubuntu-20/Makefile [hunk 3]
```diff
@@ -2,12 +2,13 @@ SHELL := bash
 
 ECR_AWS_ACCOUNT ?= $(shell aws sts get-caller-identity --profile dp-ci --query Account --output text)
 ECR_AWS_ACCOUNT_URL = $(ECR_AWS_ACCOUNT).dkr.ecr.eu-west-2.amazonaws.com
-UBUNTU_VERSION = ubuntu20.4-focal-20241011
-IMAGE_NAME = ${ECR_AWS_ACCOUNT_URL}/onsdigital/dp-concourse-tools-ubuntu-20:${UBUNTU_VERSION}
+UBUNTU_YEAR_NAME = ubuntu20.4
+UBUNTU_VERSION = focal-20241011
+IMAGE_NAME = ${ECR_AWS_ACCOUNT_URL}/onsdigital/dp-concourse-tools-ubuntu-20:${UBUNTU_YEAR_NAME}-${UBUNTU_VERSION}
```
Comment (gedge) [inline ubuntu-20/Makefile:7]:
```suggestion
IMAGE_ALIAS = onsdigital/dp-concourse-tools-ubuntu-20:${UBUNTU_YEAR_NAME}-${UBUNTU_VERSION}
IMAGE_NAME = ${ECR_AWS_ACCOUNT_URL}/${IMAGE_ALIAS}
```

File: ubuntu-20/Makefile (hunk 2 above)
Comment (gedge) [reply ubuntu-20/Makefile:11]: more consistent with the other makefile (the tag does not include the repo URL)

File: ubuntu-20/Makefile [hunk 4]
```diff
@@ -2,18 +2,19 @@ SHELL := bash
 
 ECR_AWS_ACCOUNT ?= $(shell aws sts get-caller-identity --profile dp-ci --query Account --output text)
 ECR_AWS_ACCOUNT_URL = $(ECR_AWS_ACCOUNT).dkr.ecr.eu-west-2.amazonaws.com
-UBUNTU_VERSION = ubuntu20.4-focal-20241011
-IMAGE_NAME = ${ECR_AWS_ACCOUNT_URL}/onsdigital/dp-concourse-tools-ubuntu-20:${UBUNTU_VERSION}
+UBUNTU_YEAR_NAME = ubuntu20.4
+UBUNTU_VERSION = focal-20241011
+IMAGE_ALIAS = ${ECR_AWS_ACCOUNT_URL}/onsdigital/dp-concourse-tools-ubuntu-20:${UBUNTU_YEAR_NAME}-${UBUNTU_VERSION}
```
Comment (gedge) [inline ubuntu-20/Makefile:7]:
```suggestion
IMAGE_ALIAS = onsdigital/dp-concourse-tools-ubuntu-20:${UBUNTU_YEAR_NAME}-${UBUNTU_VERSION}
```

File: ubuntu-20/Makefile [hunk 5]
```diff
@@ -2,18 +2,19 @@ SHELL := bash
...
 build:
-	docker build -t ${IMAGE_NAME} -f Dockerfile .
+	docker build -t ${IMAGE_ALIAS} --build-arg UBUNTU_VERSION=${UBUNTU_VERSION} .
 
 .PHONY: publish
 publish:
 	aws ecr get-login-password --region eu-west-2 --profile dp-ci | docker login --username AWS --password-stdin ${ECR_AWS_ACCOUNT_URL}
 
-	docker push ${IMAGE_NAME}
+	docker push ${IMAGE_ALIAS}
```
Comment (gedge) [inline ubuntu-20/Makefile:17]:
```suggestion
	docker push ${ECR_AWS_ACCOUNT_URL}/${IMAGE_ALIAS}
```

File: ubuntu-20/Makefile (hunk 3 above)
Comment (redhug1) [reply ubuntu-20/Makefile]: done

File: ubuntu-20/Makefile (hunk 4 above)
Comment (redhug1) [reply ubuntu-20/Makefile:7]: done

**Review by gedge — APPROVED** [review]

**Pull request merged**